.github/workflows/Documentation.yml

```diff
@@ -37,3 +37,11 @@ jobs:
         with:
           name: Documentation
           path: doxygen
+          
+      - name: Deploy to GitHub Pages (if on develop or main branch)
+        if: ${{ github.ref_name == 'main' || github.ref_name == 'develop' }}
```

Comment:
@JanosGit this should omit deployments of PRs

Reply:
I'd actually only do this for the `develop` branch. Otherwise the same situation occurs, the documentation either reflects the `main` or `develop` depending on what has been merged last